### PR TITLE
Switch to rust type data external checker

### PR DESCRIPTION
**Pull request by gasinvein**
Use special rust checker from flathub/flatpak-external-data-checker#93 instead of generic html checker.

### CODE REVIEW

File: org.freedesktop.Sdk.Extension.rust-stable.json [hunk 1]
```diff
@@ -33,10 +33,10 @@
                     "url": "https://static.rust-lang.org/dist/rust-1.48.0-armv7-unknown-linux-gnueabihf.tar.gz",
                     "sha256": "3aed4a63ebdd57690a31d11afbe95e6407edc224a6769be5694a1ed43bf899cb",
                     "x-checker-data": {
-                        "type": "html",
-                        "url": "https://www.rust-lang.org/",
-                        "version-pattern": "Version ((?:\\d+\\.)?(?:\\d+\\.)?\\d+)",
-                        "url-template": "https://static.rust-lang.org/dist/rust-$version-armv7-unknown-linux-gnueabihf.tar.gz"
+                        "type": "rust",
+                        "channel": "stable",
```
Comment (bilelmoussaoui) [inline org.freedesktop.Sdk.Extension.rust-stable.json]: The channel is set to stable by default

Comment (gasinvein) [reply org.freedesktop.Sdk.Extension.rust-stable.json]: I know, just thought that it's better to set it explicitly than imply.

**Comment by tilosp** [comment]
is there a reason for keeping the armv7 source in the manifest?

**Comment by gasinvein** [comment]
@tilosp Yes, it's for multilib support (mostly for symmetry with x86_64/i386), see #20

**Comment by alatiera** [comment]
I am guessing this is used by Nightly and already works?

**Comment by bilelmoussaoui** [comment]
> I am guessing this is used by Nightly and already works?

Yes, see https://github.com/flathub/org.freedesktop.Sdk.Extension.rust-nightly/pulls?q=is%3Apr+is%3Aclosed

**Pull request merged**